saige_assoc.py

```diff
@@ -200,123 +238,255 @@ def apply_job_settings(job, job_name: str):
             job.cpu(cpu)
 
 
+def run_fit_null_job(
+    null_output_path: str,
+    pheno_file: str,
+    cov_col_list: str,
+    sample_cov_col_list: str,
+    sample_id_pheno: str,
+    plink_path: str,
+    pheno_col: str,
+    skip_vre: str = 'FALSE',
+    is_cov_transform: str = 'TRUE',
+):
+    """
+    Check if the output file already exists;
+        if it does, read into a resource group
+        if it doesn't, run the command and pass the resource group back
+    Args:
+        null_output_path ():
+        pheno_file ():
+        cov_col_list ():
+        sample_cov_col_list ():
+        sample_id_pheno ():
+        plink_path ():
+        pheno_col ():
+
+    Returns:
+        Tuple: (Job | None, ResourceGroup)
+
+    """
+    if to_path(null_output_path).exists():
+        return None, get_batch().read_input_group(
+            **{
+                'rda': f'{null_output_path}.rda',
+                'varianceRatio': f'{null_output_path}.varianceRatio.txt',
+            }
+        )
+    gene_job = get_batch().new_job(name="saige-qtl")
+    gene_job.image(image_path('saige-qtl'))
+    apply_job_settings(gene_job, 'fit_null')
+
+    # create output group for first command in gene job
+    gene_job.declare_resource_group(
+        output={
+            'rda': '{root}.rda',
+            'varianceRatio': '{root}.varianceRatio.txt',
```

Comment:
same here, if the files aren't created with `.txt` extensions we should probably drop them here

Reply:
see above, I think they should be!